mentoring/participants/rest.py

```diff
@@ -25,8 +28,61 @@ class Meta:
             'comments',
         ]
 
+    def validate(self, data):
+        """Require that the `email` property, if given, matches the user during deserialization"""
+        if not self.context or "request" not in self.context:
+            raise serializers.ValidationError("validation requires request context")
+        request = self.context['request']
 
-# ViewSets define the view behavior.
-class ParticipantViewSet(viewsets.ReadOnlyModelViewSet):
+        if 'email' in data and data['email'] != request.user.email:
+            raise serializers.ValidationError({"email": "email field must match your own email"})
+
+        return data
+
+
+class IsParticipantOrAdminUser(permissions.BasePermission):
+    """
+    Object-level permission to allow users to access their own participant record.
+    """
+
+    # note that this cannot be represented as IsParticipant | IsAdminUser, as the
+    # IsAdminUser class is a view-level permission
+
+    def has_object_permission(self, request, view, obj):
+        if request.user.is_staff:
+            return True
+        if request.user.is_anonymous:
+            return False
+        return obj.email == request.user.email
+
+
+class ParticipantViewSet(
+        mixins.CreateModelMixin,
+        mixins.UpdateModelMixin,
+        viewsets.ReadOnlyModelViewSet):
     queryset = Participant.objects.all()
```

Comment:
Should this exclude ``Participants`` with ``expires`` in the past?

Reply:
The plan is to write a crontask to delete those, so I don't think that's necessary (and at least once in TC, doing so has caused bugs).